src/test/spec/unified_runner/test_event.rs

```diff
@@ -1,38 +1,191 @@
+                has_service_id: None,
             },
-            CommandEvent::Succeeded(event) => TestEvent::Succeeded {
+            CommandEvent::Succeeded(event) => ExpectedCommandEvent::Succeeded {
                 command_name: Some(event.command_name),
                 reply: Some(event.reply),
+                // TODO RUST-956 Populate the `has_service_id` field once `service_id` is present.
+                has_service_id: None,
+            },
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub enum ExpectedCmapEvent {
+    #[serde(rename = "poolCreatedEvent")]
+    PoolCreated {},
+    #[serde(rename = "poolReadyEvent")]
+    PoolReady {},
+    #[serde(rename = "poolClearedEvent", rename_all = "camelCase")]
+    PoolCleared { has_service_id: Option<bool> },
+    #[serde(rename = "poolClosedEvent")]
+    PoolClosed {},
+    #[serde(rename = "connectionCreatedEvent")]
+    ConnectionCreated {},
+    #[serde(rename = "connectionReadyEvent")]
+    ConnectionReady {},
+    #[serde(rename = "connectionClosedEvent", rename_all = "camelCase")]
+    ConnectionClosed {
+        reason: Option<ConnectionClosedReason>,
+    },
+    #[serde(rename = "connectionCheckOutStartedEvent")]
+    ConnectionCheckOutStarted {},
+    #[serde(rename = "connectionCheckOutFaildEvent", rename_all = "camelCase")]
+    ConnectionCheckOutFailed {
+        reason: Option<ConnectionCheckoutFailedReason>,
+    },
+    #[serde(rename = "connectionCheckedOutEvent")]
+    ConnectionCheckedOut {},
+    #[serde(rename = "connectionCheckedInEvent")]
+    ConnectionCheckedIn {},
+}
+
+impl From<CmapEvent> for ExpectedCmapEvent {
+    fn from(event: CmapEvent) -> Self {
+        match event {
+            CmapEvent::PoolCreated(_) => Self::PoolCreated {},
+            CmapEvent::PoolClosed(_) => Self::PoolClosed {},
+            CmapEvent::PoolReady(_) => Self::PoolReady {},
+            CmapEvent::ConnectionCreated(_) => Self::ConnectionCreated {},
+            CmapEvent::ConnectionReady(_) => Self::ConnectionReady {},
+            CmapEvent::ConnectionClosed(ev) => Self::ConnectionClosed {
+                reason: Some(ev.reason),
             },
+            CmapEvent::ConnectionCheckOutStarted(_) => Self::ConnectionCheckOutStarted {},
+            CmapEvent::ConnectionCheckOutFailed(ev) => Self::ConnectionCheckOutFailed {
+                reason: Some(ev.reason),
+            },
+            CmapEvent::ConnectionCheckedOut(_) => Self::ConnectionCheckedOut {},
+            CmapEvent::PoolCleared(_) => {
+                // TODO RUST-956 populate `has_service_id`
+                Self::PoolCleared {
+                    has_service_id: None,
+                }
+            }
+            CmapEvent::ConnectionCheckedIn(_) => Self::ConnectionCheckedIn {},
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Deserialize)]
+pub enum ObserveEvent {
```

Comment:
I opted to keep this a single `enum` rather than splitting into `ObserveCommandEvent` / `ObserveCmapEvent` to keep this as close to the spec descriptions as possible.